Gems/AtomLyIntegration/EditorModeFeedback/Code/Source/Pass/EditorStatePassSystem.cpp

```diff
@@ -49,107 +49,118 @@ namespace AZ::Render
     void EditorStatePassSystem::AddPassesToRenderPipeline(RPI::RenderPipeline* renderPipeline)
     {
         const auto templateName = Name(MainPassParentTemplateName);
-        if (RPI::PassSystemInterface::Get()->GetPassTemplate(templateName))
-        {
-            // Template was created by another pipeline, do not to create again
-            return;
-        }
 
-        auto mainParentPassTemplate = AZStd::make_shared<RPI::PassTemplate>();
-        mainParentPassTemplate->m_name = templateName;
-        mainParentPassTemplate->m_passClass = Name(MainPassParentTemplatePassClassName);
-        
-        // Input depth slot
-        {
-            RPI::PassSlot slot;
-            slot.m_name = Name("InputDepth");
-            slot.m_slotType = RPI::PassSlotType::Input;
-            mainParentPassTemplate->AddSlot(slot);
-        }
-        
-        // Input/output color slot
+        // Early return if pass is already found in render pipeline.
+        auto passFilter = AZ::RPI::PassFilter::CreateWithTemplateName(templateName, renderPipeline);
+        auto foundPass = AZ::RPI::PassSystemInterface::Get()->FindFirstPass(passFilter);
+        if (foundPass)
         {
-            RPI::PassSlot slot;
-            slot.m_name = Name("ColorInputOutput");
-            slot.m_slotType = RPI::PassSlotType::InputOutput;
-            mainParentPassTemplate->AddSlot(slot);
+            return;
         }
 
-        // Entity mask passes
-        m_masks = CreateMaskPassTemplatesFromEditorStates(m_editorStates);
-        for (const auto& drawList : m_masks)
+        auto mainParentPassTemplate = RPI::PassSystemInterface::Get()->GetPassTemplate(templateName);
+        if (!mainParentPassTemplate)
         {
-            RPI::PassRequest pass;
-            pass.m_passName = GetMaskPassNameForDrawList(drawList);
-            pass.m_templateName = GetMaskPassTemplateNameForDrawList(drawList);
+            // Create the pass template and add it to the pass system.
+            auto newPassTemplate = AZStd::make_shared<RPI::PassTemplate>();
```

Comment:
All the code inside this block is exactly the same but using `newPassTemplate` variable and indented. Slack didn't pick up the changes very well.